c7n/resources/elasticache.py

```diff
@@ -472,3 +473,48 @@ class resource_type(TypeInfo):
         cfn_type = 'AWS::ElastiCache::ReplicationGroup'
 
     permissions = ('elasticache:DescribeReplicationGroups',)
+
+
+@ElastiCacheReplicationGroup.filter_registry.register('kms-key')
+class KmsFilter(KmsRelatedFilter):
+
+    RelatedIdsExpression = 'KmsKeyId'
+
+
+@ElastiCacheReplicationGroup.action_registry.register('delete')
+class DeleteReplicationGroup(BaseAction):
+    """Action to delete a cache replication group
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: elasticache-delete-replication-group
+                resource: elasticache-group
+                filters:
+                  - type: value
+                    key: AtRestEncryptionEnabled
+                    value: False
+                actions:
+                  - type: delete
+                    snapshot: False
+
+    """
+    schema = type_schema(
+        'delete', **{'snapshot': {'type': 'boolean'}})
+
+    valid_origin_states = ('available',)
+    permissions = ('elasticache:DeleteReplicationGroup',)
+
+    def process(self, resources):
+        resources = self.filter_resources(resources, 'Status', self.valid_origin_states)
+        client = local_session(self.manager.session_factory).client('elasticache')
+        for r in resources:
+            params = {'ReplicationGroupId': r['ReplicationGroupId']}
+            if self.data.get('snapshot', False):
+                params.update({'FinalSnapshotIdentifier': r['ReplicationGroupId'] + '-snapshot'})
+            self.manager.retry(client.delete_replication_group, **params, ignore_err_codes=(
+                'ReplicationGroupNotFoundFault',
+                'InvalidParameterValueException', 'InvalidParameterCombinationException'
+                'SnapshotFeatureNotSupportedFault'))
```

Comment:
I feel I should take out these three exceptions from here: 'InvalidParameterValueException',  'InvalidParameterCombinationException',  'SnapshotFeatureNotSupportedFault' so that they can bubble up and not fail silently

Reply:
Agreed, it seems odd we'd even expect a InvalidParameterCombinationException, those could probably be prevented from the start in the validate method.